src/runtime/parser.c

```diff
@@ -239,34 +269,27 @@ static Tree *parser__lex(Parser *self, TSStateId parse_state) {
         self->lexer.data.result_symbol = ts_builtin_sym_error;
         break;
       }
-      self->lexer.data.advance(&self->lexer, ERROR_STATE, false);
+      self->lexer.data.advance(&self->lexer, false);
     }
 
     skipped_error = true;
     error_end_position = self->lexer.current_position;
   }
 
   Tree *result;
-
   if (skipped_error) {
     Length padding = length_sub(error_start_position, start_position);
     Length size = length_sub(error_end_position, error_start_position);
     ts_lexer_reset(&self->lexer, error_end_position);
     result = ts_tree_make_error(size, padding, first_error_character);
   } else {
     TSSymbol symbol = self->lexer.data.result_symbol;
-    Length padding =
-      length_sub(self->lexer.token_start_position, start_position);
-    Length size = length_sub(self->lexer.current_position,
-                                  self->lexer.token_start_position);
-    result =
-      ts_tree_make_leaf(symbol, padding, size,
-                        ts_language_symbol_metadata(self->language, symbol));
+    Length padding = length_sub(self->lexer.token_start_position, start_position);
+    Length size = length_sub(self->lexer.current_position, self->lexer.token_start_position);
+    TSSymbolMetadata metadata = ts_language_symbol_metadata(self->language, symbol);
+    result = ts_tree_make_leaf(symbol, padding, size, metadata);
   }
 
-  if (!result)
```

Comment:
Yeah, this conditional was leftover from before https://github.com/tree-sitter/tree-sitter/pull/43, when the system tried to recover from malloc failure. I've removed that logic across the board, since malloc never even actually returns null on most systems, and apps that use tree-sitter aren't likely to have a fine-grained OOM recovery strategy anyway.